jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcUpdateCallback.java

```diff
@@ -63,10 +63,12 @@ public void executePreparedStatement(PreparedStatement preparedStatement, boolea
     protected final Connection getConnection() {
         if (_connection == null) {
             _connection = getDataContext().getConnection();
-            try {
-                _connection.setAutoCommit(false);
-            } catch (SQLException e) {
-                throw JdbcUtils.wrapException(e, "disable auto-commit");
+            if (_dataContext.getQueryRewriter().isTransactional()) {
+                try {
```

Comment:
At least it's what I meant :-) We only have Hive in our portfolio currently that is a "non transactional" JDBC driver. In the case of Hive it is not allowed to call these methods.

Do you have an underlying concern with this specifically?